### PR TITLE
Add support for codeowners to repo

### PR DESCRIPTION
**Pull request by ryanbogan**
Signed-off-by: Ryan Bogan <rbogan@amazon.com>

### Description
Adding a codeowners file.

### Issues resolved
Progress on:
[Meta issue](https://github.com/opensearch-project/project-meta/issues/31)
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by ylwu-amzn** [comment]
The commit message "initial commit" looks not match what this PR changed, can you tune the commit message and explain what this PR for?

**Review by CEHENKLE — APPROVED** [review]

**Review by ohltyler — APPROVED** [review]
Thanks for adding! Please rename the PR before squashing and merging to keep a cleaner commit history.

**Comment by dblock** [comment]
@ryanbogan You also will need to amend the message in the actual commit, please.

**Comment by ryanbogan** [comment]
> @ryanbogan You also will need to amend the message in the actual commit, please.

Done

**Review by CEHENKLE — APPROVED** [review]

**Review by ohltyler — APPROVED** [review]

**Comment by CEHENKLE** [comment]
@opensearch-project/anomaly-detection I've updated the branch rules to use this codeowner file. What do you want to set the number of approvals required to merge a PRs as? 1 or 2?

At some point we may require consistency, but I don't think the direction is clear right now. So since I'm having to go in and update the rules, if you have a preference, LMK.

**Comment by ohltyler** [comment]
@CEHENKLE I think 1 approval is preferred to prevent being blocked on merging urgent PRs, such as backporting things to a release branch. But no strong preference.

**Pull request merged**
